### PR TITLE
[Feature] Signature px payment method selector

### PR DESCRIPTION
**Pull request by jorGonzalez291292**
## Motivación y Contexto
Checkout becomes a payment method selector.

## Descripción
A new signature is implemented to start the checkout as a payment method selector. We can start checkout.

### CODE REVIEW

File: px-checkout/src/main/java/com/mercadopago/android/px/configuration/PaymentConfiguration.kt [hunk 1]
```diff
@@ -13,42 +13,64 @@ import com.mercadopago.android.px.core.internal.PaymentProcessorMapper
...
-        builder.charges, builder.paymentProcessor, builder.paymentProcessorV2
+    private constructor(builder: Builder) : this(
+        builder.charges, builder.paymentProcessor, builder.paymentProcessorV2, builder.supportSplit
     )
 
     constructor(parcel: Parcel) : this(
         parcel.createTypedArrayList(PaymentTypeChargeRule.CREATOR)!!,
         parcel.readParcelable(SplitPaymentProcessor::class.java.classLoader)!!,
-        parcel.readParcelable(PaymentProcessorV2::class.java.classLoader)!!
+        parcel.readParcelable(PaymentProcessorV2::class.java.classLoader)!!,
```
Comment (guchito9) [inline px-checkout/src/main/java/com/mercadopago/android/px/configuration/PaymentConfiguration.kt:33]: It's nullable this field now, we shouldn't force it with !!

File: px-checkout/src/main/java/com/mercadopago/android/px/configuration/PaymentConfiguration.kt [hunk 2]
```diff
@@ -13,42 +13,64 @@ import com.mercadopago.android.px.core.internal.PaymentProcessorMapper
...
 
     @Deprecated("")
-    val discountConfiguration:DiscountConfiguration? = null
+    val discountConfiguration: DiscountConfiguration? = null
 
     @Deprecated("")
-    val paymentMethodPluginList:Collection<PaymentMethodPlugin> = ArrayList()
+    val paymentMethodPluginList: Collection<PaymentMethodPlugin> = ArrayList()
 
     internal fun getCheckoutType(): CheckoutType {
```
Comment (guchito9) [inline px-checkout/src/main/java/com/mercadopago/android/px/configuration/PaymentConfiguration.kt:43]: We should discuss what to do with this, cause right now, with no payment processor it returns CUSTOM_REGULAR

File: px-checkout/src/main/java/com/mercadopago/android/px/core/MercadoPagoCheckout.kt [hunk 3]
```diff
@@ -77,6 +80,7 @@ class MercadoPagoCheckout internal constructor(builder: Builder) {
         val publicKey: String
         val preferenceId: String?
         val checkoutPreference: CheckoutPreference?
+        internal var scheduledPaymentMethodType = ScheduledPaymentMethodType.NON_SCHEDULED
```
Comment (guchito9) [inline px-checkout/src/main/java/com/mercadopago/android/px/core/MercadoPagoCheckout.kt:83]: Shouldn't ALL be the default value?

File: px-checkout/src/main/java/com/mercadopago/android/px/core/PXPaymentMethodSelector.kt [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+    }
+
+    class Builder(val publicKey: String, val preferenceId: String) {
+
+        internal lateinit var accessToken: String
+        internal var dynamicDialogConfiguration = DynamicDialogConfiguration.Builder().build()
+        internal var customStringConfiguration = CustomStringConfiguration.Builder().build()
+        internal var discountParamsConfiguration = DiscountParamsConfiguration.Builder().build()
+        internal var trackingConfiguration = TrackingConfiguration.Builder().build()
+        internal var scheduledPaymentMethodType = ScheduledPaymentMethodType.NON_SCHEDULED
```
Comment (guchito9) [inline px-checkout/src/main/java/com/mercadopago/android/px/core/PXPaymentMethodSelector.kt:61]: The same as before. I know that the first integration that uses this feature requires non scheduled, but the default value for us should be all.

**Comment by jorGonzalez291292** [comment]
@guchito9 I had put everything that has to do with ScheduledPaymentMethodType as a proposal, but before I left I changed this a bit. I'm sure I'll change to something else.

File: px-checkout/src/main/java/com/mercadopago/android/px/configuration/PaymentConfiguration.kt [hunk 5]
```diff
@@ -85,6 +103,10 @@ class PaymentConfiguration private constructor(val charges: ArrayList<PaymentTyp
             this.charges.addAll(charges)
         }
 
+        internal fun setSupportSplit(supportSplit: Boolean) = apply {
```
Comment (guchito9) [inline px-checkout/src/main/java/com/mercadopago/android/px/configuration/PaymentConfiguration.kt:106]: Shouldn't this be public so the integrator can use it?

Comment (jorGonzalez291292) [reply px-checkout/src/main/java/com/mercadopago/android/px/configuration/PaymentConfiguration.kt:106]: Today the way that the integrator has to define if split is supported is through the payment processor. I did not want to lend to confusion, just for that reason I left it as internal.

File: px-checkout/src/main/java/com/mercadopago/android/px/configuration/PaymentConfiguration.kt [hunk 6]
```diff
@@ -13,42 +13,61 @@ import com.mercadopago.android.px.core.internal.PaymentProcessorMapper
...
-        return when(paymentProcessorV2) {
+        return when (paymentProcessorV2) {
             is ScheduledPaymentProcessor -> CheckoutType.CUSTOM_SCHEDULED
             is DefaultPaymentProcessor -> CheckoutType.DEFAULT_REGULAR
             else -> CheckoutType.CUSTOM_REGULAR
         }
     }
 
+
+    fun getSupportsSplit(checkoutPreference: CheckoutPreference?) : Boolean {
```
Comment (guchito9) [inline px-checkout/src/main/java/com/mercadopago/android/px/configuration/PaymentConfiguration.kt:52]: Hay chances de que sea internal esto?

**Pull request merged**